src/agent/rustjail/src/container.rs

```diff
@@ -924,14 +931,24 @@ impl BaseContainer for LinuxContainer {
 
         if p.init {
             let spec = self.config.spec.as_mut().unwrap();
-            update_namespaces(&self.logger, spec, p.pid)?;
+            update_namespaces(spec, p.pid)?;
         }
         self.processes.insert(p.pid, p);
 
         info!(logger, "wait on child log handler");
-        log_handler.join();
+        match log_handler.join() {
```

Comment:
Since `Ok` is ignored, can we change this to:

```rust
if let Err(e) = log_handler.join() {
    return Err(ErrorKind::ErrorCode(format!("Failed joining with child handler {:?}", e)).into())
}
```

 ?